### PR TITLE
LPD-42768 label:for and input:id don't match in Input Localized taglib when using special characters

### PR DESCRIPTION
**Pull request by antonio-ortega**
Hi,

Right now, when using `aui:input` taglib with a name containing special characters makes `label`'s for attribute and `input`'s id attribute don't match. This itself is already wrong, but apart from that is causing issues, for example, when using AutoFields as it is explained in [LPD-42768](https://liferay.atlassian.net/browse/LPD-42768).

This is happening because in [LPS-43171](https://liferay.atlassian.net/browse/LPSA-27247) we made `input`'s id `AUICompatible`:
`String id = HtmlUtil.getAUICompatibleId((String)request.getAttribute("liferay-ui:input-localized:id"));`

That was needed because you could create a custom field whose key could cause XSS. However, now we are checking that on creation, not letting the user introduce forbidden character when adding a new field, so this transformation is not needed anymore.

IMPORTANT: In input_localized.jsp I've added following line:
`<aui:input label="" localized="<%= true %>" name="input-localized-name" type="text" value="" />`

**That name attribute is not following our pattern**, which is camelCase. This is intentional, since I need to add some special character, such as a hyphen, to test this issue.

Let me know if you have any doubt.

Thanks.

### CODE REVIEW

**Comment by antonio-ortega** [comment]
ci:test:sf

**Comment by antonio-ortega** [comment]
ci:test:relevant

**Comment by antonio-ortega** [comment]
ci:test:sf

**Comment by antonio-ortega** [comment]
ci:test:relevant

**Review by dsanz — APPROVED** [review]
LGTM ;) just see inline comments

File: modules/apps/frontend-taglib/frontend-taglib-sample-web/src/main/resources/META-INF/resources/partials/input_localized.jsp [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+--%>
+
+<%@ include file="/init.jsp" %>
+
+<clay:container-fluid>
+	<aui:input label="" localized="<%= true %>" name="input-localized-name" type="text" value="" />
```
[the remark this answers is not in the document]
Comment (antonio-ortega) [reply modules/apps/frontend-taglib/frontend-taglib-sample-web/src/main/resources/META-INF/resources/partials/input_localized.jsp:11]: Thanks!!

**Comment by antonio-ortega** [comment]
ci:forward

**Pull request closed without merging**